### ISSUE TITLE
Upgrade MUI-X DataGrid to v7

### ISSUE THREAD

**Issue by marcellmueller** [bcgov/cas-registration#1410]
###Description of the Tech Debt###

While working on the Accessibility testing and DataGrid search filters I tried using the recently released [DataGrid V7](https://github.com/mui/mui-x/releases/tag/v7.2.0). It fails our axe accessibility test so creating this ticket to track it and hope it's fixed in the future. Initially I thought it might be our implementation of it but installing the Axe chrome plugin and running the scan on the MUI DataGrid example page shows the same failure.

###Tech Debt Triage#
The purpose of our technical debt triage process is to analyze technical debt to determine risk level of the technical debt and the value in tackling that technical debt.

Risk Value Scoring:

| Level  | Value |
| ------ | ----- |
| High   | 3     |
| Medium | 2     |
| Low    | 1     |

| Technical Debt - Risk Types                                                                                                   | Level | Value |
| ----------------------------------------------------------------------------------------------------------------------------- | ----- | ----- |
| Business Area Risk - Risk of business area visibility / damage to user experience                                             |      2 |       |
| Developer Fault Risk - How likely will this tech debt cause a future error related to coding on top of it                     |   2    |       |
| System Fault Risk - Risk of system errors or application downtime                                                             |    2  |       |
| Time Scale Risk - Compound risk effect if left alone. How much more difficult to fix or dangerous will this become over time? |   3    |       |
| Time Sink Risk - How much will this tech debt slow the development process down                                               |   2    |       |
| TOTAL SCORE:                                                                                                                  |  11     |       |

**Comment by andrea-williams** [comment]
July 2025: we're still using v6 of the MUI Datagrid

**Comment by Sepehr-Sobhani** [comment]
@patriciarussellCAS Ticket #4634 also covers this, so we can close this one.
